scipy/stats/_multivariate.py

```diff
@@ -3448,16 +3448,17 @@ def rvs(self, dim, size=1, random_state=None):
             Random size N-dimensional matrices, dimension (size, dim, dim)
 
         """
+        random_state = self._get_random_state(random_state)
+
         size = int(size)
         if size > 1:
             return np.array([self.rvs(dim, size=1, random_state=random_state)
                              for i in range(size)])
 
         dim = self._process_parameters(dim)
 
-        random_state = self._get_random_state(random_state)
-
         H = np.eye(dim)
+        H[-1, -1] = (-1)**random_state.randint(2)
```

Comment:
This is multiplying by \tilde{H}_1^T from the *wrong side*. See Eq. (7.7) in Mezzadri. You may also want to think if extending the loop index to `dim+1` does the same thing.

Reply:
Also, in the loop below, please fix
```
D = np.sign(x[0]) if x[0] != 0 else 1
```